core/src/consensus/tendermint/worker.rs

```diff
@@ -307,16 +304,6 @@ impl Worker {
                             }) => {
                                 inner.set_signer(ap, address);
                             }
-                            Ok(Event::AllowedHeight {
-                                result,
-                            }) => {
-                                let allowed_height = if inner.step.is_commit() {
-                                    inner.height + 1
```

Comment:
This was the cause of a bug.
If a proposal is not imported in Commit state yet, the code should return `inner.height` instead of `inner.height + 1`.